crates/rspack_core/src/concatenated_module.rs

```diff
@@ -1989,10 +1989,41 @@ impl ConcatenatedModule {
       let comments = SwcComments::default();
       let mut module_info = concatenation_scope.current_module;
 
+      let jsx = {
+        let get_jsx = |options: &ParserOptions| -> Option<bool> {
+          match options {
+            ParserOptions::Javascript(js_options)
+            | ParserOptions::JavascriptAuto(js_options)
+            | ParserOptions::JavascriptEsm(js_options)
+            | ParserOptions::JavascriptDynamic(js_options) => js_options.jsx,
+            _ => None,
+          }
+        };
+
+        module
+          .as_ref()
+          .as_normal_module()
+          .and_then(|normal_module| normal_module.get_parser_options())
+          .and_then(get_jsx)
+          .or_else(|| {
+            compilation
+              .options
+              .module
+              .parser
```

Comment:
You don't need to read from compilation.options here if it's None, normal module will always has parser options once it's factorized, if it's None, it means the module is not normal module, you can just use `unwarp_or(false)` here, because the codegen of non-normal module won't contains jsx

Reply:
PTAL, i think i made it right this time. 😇

Reply:
thank u Kun!